pkg/frontend/v1/queue_test.go

```diff
@@ -39,194 +49,118 @@ func TestDequeuesExpiredRequests(t *testing.T) {
 	flagext.DefaultValues(&config)
 	config.MaxOutstandingPerTenant = 10
 	userID := "1"
-	userID2 := "2"
 
-	f, err := setupFrontend(config)
+	f, err := setupFrontend(t, config)
 	require.NoError(t, err)
 
 	ctx := user.InjectOrgID(context.Background(), userID)
 	expired, cancel := context.WithCancel(ctx)
 	cancel()
 
+	good := 0
 	for i := 0; i < config.MaxOutstandingPerTenant; i++ {
 		var err error
 		if i%5 == 0 {
-			err = f.queueRequest(ctx, testReq(ctx))
+			good++
+			err = f.queueRequest(ctx, testReq(ctx, fmt.Sprintf("good-%d", i), userID))
 		} else {
-			err = f.queueRequest(ctx, testReq(expired))
+			err = f.queueRequest(ctx, testReq(expired, fmt.Sprintf("expired-%d", i), userID))
 		}
 
 		require.Nil(t, err)
 	}
 
-	// the first request shouldn't be expired
-	req, idx, err := f.getNextRequestForQuerier(ctx, -1, "")
-	require.Nil(t, err)
-	require.NotNil(t, req)
-	require.Equal(t, 9, len(f.queues.getOrAddQueue(userID, 0)))
-
-	// the next unexpired request should be the 5th index
-	req, idx, err = f.getNextRequestForQuerier(ctx, idx, "")
-	require.Nil(t, err)
-	require.NotNil(t, req)
-	require.Equal(t, 4, len(f.queues.getOrAddQueue(userID, 0)))
-
-	// add one request to a second tenant queue
-	ctx2 := user.InjectOrgID(context.Background(), userID2)
-	err = f.queueRequest(ctx2, testReq(ctx2))
-	require.Nil(t, err)
-
-	// there should be no more unexpired requests in queue until the second tenant enqueues one.
-	req, _, err = f.getNextRequestForQuerier(ctx, idx, "")
-	require.Nil(t, err)
-	require.NotNil(t, req)
-
-	// ensure either one or two queues are fully drained, depending on which was requested first
-	_, ok := f.queues.userQueues[userID]
-	if ok {
-		// if the second user's queue was chosen for the last request,
-		// the first queue should still contain 4 (expired) requests.
-		require.Equal(t, 4, len(f.queues.getOrAddQueue(userID, 0)))
+	// Replay entire conversation in one second.
```

Comment:
I find this comment unclear. Is there a better way to phrase it?

Reply:
I've tried:

```
	// Calling Process will only return when client disconnects or context is finished.
	// We use context timeout to stop Process call.
```

Reply:
Way more clear, thanks!